assets/js/base/context/providers/cart-checkout/checkout-state/utils.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * Internal dependencies
+ */
+import type { PaymentResultDataType, CheckoutResponse } from './types';
+
+/**
+ * Prepares the payment_result data from the server checkout endpoint response.
+ */
+export const getPaymentResultFromCheckoutResponse = (
+	response: CheckoutResponse
+): PaymentResultDataType => {
+	const paymentResult = {
+		message: '',
+		paymentStatus: '',
+		redirectUrl: '',
+		paymentDetails: {},
+	} as PaymentResultDataType;
+
+	// payment_result is present in successful responses.
+	if ( 'payment_result' in response ) {
```

Comment:
Would it be wise to add additional checks about what `response` is here? Or are we always guaranteed an object?

Reply:
It appears to always be an object—everywhere I see usage of `setAfterProcessing` it's dealing with objects. Those files are not typed yet.